File: compare_matlab_python_transmission.m
% Compare MATLAB and Python atmospheric transmission implementations

% Create identical test conditions
Config = transmission.inputConfig('default');

% Verify we're using the same parameters as Python
fprintf('=== MATLAB ATMOSPHERIC TRANSMISSION ===\n\n');
fprintf('Test conditions:\n');
fprintf('  Zenith angle: %.1f deg\n', Config.Atmospheric.Zenith_angle_deg);
fprintf('  Pressure: %.1f mbar\n', Config.Atmospheric.Pressure_mbar);
fprintf('  Temperature: %.1f C\n', Config.Atmospheric.Temperature_C);
fprintf('  Ozone: %.0f DU\n', Config.Atmospheric.Components.Ozone.Dobson_units);
fprintf('  Water vapor: %.1f cm\n', Config.Atmospheric.Components.Water.Pwv_cm);
fprintf('  Aerosol AOD: %.3f\n', Config.Atmospheric.Components.Aerosol.Tau_aod500);
fprintf('  CO2: %.0f ppm\n', Config.Atmospheric.Components.Molecular_absorption.Co2_ppm);

% Create identical wavelength array - 101 points from 300 to 1100 nm
wavelengths = linspace(300, 1100, 101)';

% Calculate total transmission
fprintf('\nCalculating components...\n');
Trans_total = transmission.atmosphericTransmission(wavelengths, Config);

% Also calculate individual components for detailed comparison
% Rayleigh only
Config_ray = Config;
Config_ray.Atmospheric.Components.Ozone.Enable = false;
Config_ray.Atmospheric.Components.Water.Enable = false;
Config_ray.Atmospheric.Components.Aerosol.Enable = false;
Config_ray.Atmospheric.Components.Molecular_absorption.Enable = false;
Trans_ray = transmission.atmosphericTransmission(wavelengths, Config_ray);

% Ozone only
Config_oz = Config;
Config_oz.Atmospheric.Components.Rayleigh.Enable = false;
Config_oz.Atmospheric.Components.Water.Enable = false;
Config_oz.Atmospheric.Components.Aerosol.Enable = false;
Config_oz.Atmospheric.Components.Molecular_absorption.Enable = false;
Trans_oz = transmission.atmosphericTransmission(wavelengths, Config_oz);

% Water only
Config_water = Config;
Config_water.Atmospheric.Components.Rayleigh.Enable = false;
Config_water.Atmospheric.Components.Ozone.Enable = false;
Config_water.Atmospheric.Components.Aerosol.Enable = false;
Config_water.Atmospheric.Components.Molecular_absorption.Enable = false;
Trans_water = transmission.atmosphericTransmission(wavelengths, Config_water);

% Aerosol only
Config_aer = Config;
Config_aer.Atmospheric.Components.Rayleigh.Enable = false;
Config_aer.Atmospheric.Components.Ozone.Enable = false;
Config_aer.Atmospheric.Components.Water.Enable = false;
Config_aer.Atmospheric.Components.Molecular_absorption.Enable = false;
Trans_aer = transmission.atmosphericTransmission(wavelengths, Config_aer);

% UMG only
Config_umg = Config;
Config_umg.Atmospheric.Components.Rayleigh.Enable = false;
Config_umg.Atmospheric.Components.Ozone.Enable = false;
Config_umg.Atmospheric.Components.Water.Enable = false;
Config_umg.Atmospheric.Components.Aerosol.Enable = false;
Trans_umg = transmission.atmosphericTransmission(wavelengths, Config_umg);

% Output results in same format as Python
fprintf('\n# Wavelength(nm), Trans_Total, Trans_Ray, Trans_Oz, Trans_Water, Trans_Aer, Trans_UMG\n');
for i = 1:length(wavelengths)
    fprintf('%.1f, %.6f, %.6f, %.6f, %.6f, %.6f, %.6f\n', ...
        wavelengths(i), Trans_total(i), Trans_ray(i), Trans_oz(i), ...
        Trans_water(i), Trans_aer(i), Trans_umg(i));
end

% Summary statistics
fprintf('\n=== SUMMARY ===\n');
fprintf('Mean total transmission: %.4f\n', mean(Trans_total));
fprintf('Min total transmission: %.4f\n', min(Trans_total));
fprintf('Max total transmission: %.4f\n', max(Trans_total));

% Band averages
uv_mask = wavelengths >= 300 & wavelengths <= 400;
vis_mask = wavelengths >= 400 & wavelengths <= 700;
nir_mask = wavelengths >= 700 & wavelengths <= 1100;

fprintf('\nMean UV transmission (300-400 nm): %.4f\n', mean(Trans_total(uv_mask)));
fprintf('Mean visible transmission (400-700 nm): %.4f\n', mean(Trans_total(vis_mask)));
fprintf('Mean NIR transmission (700-1100 nm): %.4f\n', mean(Trans_total(nir_mask)));

% Save results for comparison
save('matlab_transmission_results.mat', 'wavelengths', 'Trans_total', ...
     'Trans_ray', 'Trans_oz', 'Trans_water', 'Trans_aer', 'Trans_umg');

File: compare_matlab_python_transmission.py
#!/usr/bin/env python3
"""
Compare Python and MATLAB atmospheric transmission implementations
"""
import numpy as np
import matplotlib.pyplot as plt
import sys
sys.path.append('/home/dana/anaconda3/envs/myenv/lib/python3.12/site-packages')

from transmission_fitter.atmospheric_models import (
    Rayleigh_Transmission, 
    Ozone_Transmission,
    WaterTransmittance,
    Aerosol_Transmission,
    UMGTransmittance
)

# Standard test conditions matching MATLAB defaults
z_ = 0.0           # Zenith angle (degrees)
p_ = 1013.25       # Pressure (mbar)
tair = 15.0        # Temperature (C)
dobson = 300       # Ozone (Dobson units)
pwv = 1.0          # Precipitable water vapor (cm)
tau_aod = 0.084    # Aerosol optical depth at 500nm
alpha = 0.6        # Angstrom exponent
co2_ppm = 395      # CO2 concentration

# Create wavelength array - 101 points from 300 to 1100 nm
wavelengths = np.linspace(300, 1100, 101)

print("=== PYTHON ATMOSPHERIC TRANSMISSION ===")
print(f"\nTest conditions:")
print(f"  Zenith angle: {z_} deg")
print(f"  Pressure: {p_} mbar")
print(f"  Temperature: {tair} C")
print(f"  Ozone: {dobson} DU")
print(f"  Water vapor: {pwv} cm")
print(f"  Aerosol AOD: {tau_aod}")
print(f"  CO2: {co2_ppm} ppm")

# Calculate individual components
print("\nCalculating components...")

# Rayleigh
ray = Rayleigh_Transmission(z_, p_)
trans_ray = ray.make_transmission()
trans_ray_interp = np.interp(wavelengths, ray.wvl_arr, trans_ray)

# Ozone
oz = Ozone_Transmission(z_, dobson)
trans_oz = oz.make_transmission()
trans_oz_interp = np.interp(wavelengths, oz.wvl_arr, trans_oz)

# Water
water = WaterTransmittance(z_, pwv, p_)
trans_water = water.make_transmission()
trans_water_interp = np.interp(wavelengths, water.wvl_arr, trans_water)

# Aerosol
aer = Aerosol_Transmission(z_, tau_aod, alpha)
trans_aer = aer.make_transmission()
trans_aer_interp = np.interp(wavelengths, aer.wvl_arr, trans_aer)

# UMG with trace gases
umg = UMGTransmittance(z_, tair, p_, co2_ppm, with_trace_gases=True)
trans_umg = umg.make_transmission()
trans_umg_interp = np.interp(wavelengths, umg.wvl_arr, trans_umg)

# Total transmission
trans_total = trans_ray_interp * trans_oz_interp * trans_water_interp * trans_aer_interp * trans_umg_interp

# Output results for comparison
print("\n# Wavelength(nm), Trans_Total, Trans_Ray, Trans_Oz, Trans_Water, Trans_Aer, Trans_UMG")
for i in range(len(wavelengths)):
    print(f"{wavelengths[i]:.1f}, {trans_total[i]:.6f}, {trans_ray_interp[i]:.6f}, "
          f"{trans_oz_interp[i]:.6f}, {trans_water_interp[i]:.6f}, "
          f"{trans_aer_interp[i]:.6f}, {trans_umg_interp[i]:.6f}")

plt.figure()
plt.plot(wavelengths, trans_total);
plt.show()

# Summary statistics
print(f"\n=== SUMMARY ===")
print(f"Mean total transmission: {np.mean(trans_total):.4f}")
print(f"Min total transmission: {np.min(trans_total):.4f}")
print(f"Max total transmission: {np.max(trans_total):.4f}")

# Band averages
uv_mask = (wavelengths >= 300) & (wavelengths <= 400)
vis_mask = (wavelengths >= 400) & (wavelengths <= 700)
nir_mask = (wavelengths >= 700) & (wavelengths <= 1100)

print(f"\nMean UV transmission (300-400 nm): {np.mean(trans_total[uv_mask]):.4f}")
print(f"Mean visible transmission (400-700 nm): {np.mean(trans_total[vis_mask]):.4f}")
print(f"Mean NIR transmission (700-1100 nm): {np.mean(trans_total[nir_mask]):.4f}")